client/core/core.go

```diff
@@ -1455,22 +1455,54 @@ func (c *Core) connectAndUpdateWallet(w *xcWallet) error {
 	return nil
 }
 
+// reloadWallet creates a fresh xcWallet via loadWallet from the provided
+// xcWallet. This creates a new asset.Wallet so that repeated connection
+// attempts are not met with confusing "already connected" errors if the
+// underlying asset.Wallet implementation is not capable of handling repeated
+// Connect request, which is a common limitation of wallet clients, especially
+// after a Disconnect.
+func (c *Core) reloadWallet(wallet *xcWallet) (*xcWallet, error) {
+	wallet.mtx.RLock()
+	defer wallet.mtx.RUnlock()
+	return c.loadWallet(&db.Wallet{
+		AssetID:     wallet.AssetID,
+		Type:        wallet.cfg.Type,
+		Settings:    wallet.cfg.Settings,
+		Balance:     wallet.balance.Balance,
+		EncryptedPW: wallet.encPass,
+		Address:     wallet.address,
+	})
+}
+
 // connectedWallet fetches a wallet and will connect the wallet if it is not
 // already connected. If the wallet gets connected, this also emits WalletState
 // and WalletBalance notification.
 func (c *Core) connectedWallet(assetID uint32) (*xcWallet, error) {
-	wallet, exists := c.wallet(assetID)
+	c.walletMtx.Lock() // may replace the xcWallet in the map if connect fails
+	defer c.walletMtx.Unlock()
+	wallet, exists := c.wallets[assetID]
 	if !exists {
 		return nil, newError(missingWalletErr, "no configured wallet found for %s (%d)",
 			strings.ToUpper(unbip(assetID)), assetID)
 	}
-	if !wallet.connected() {
-		err := c.connectAndUpdateWallet(wallet)
-		if err != nil {
-			return nil, err
-		}
+	if wallet.connected() {
+		return wallet, nil
+	}
+
+	err := c.connectAndUpdateWallet(wallet)
+	if err == nil {
+		return wallet, nil
 	}
-	return wallet, nil
+
+	// When connect fails, assume the wallet is dead, so create a new xcWallet.
+	if newWallet, errLoad := c.reloadWallet(wallet); errLoad == nil {
+		c.wallets[assetID] = newWallet
+	} else {
+		// Leave the old xcWallet in the map for future attempt.
+		c.log.Errorf("Failed to reload %v wallet: %v", unbip(assetID), errLoad)
+	}
```

Comment:
My instinct is to formalize this as a policy and make asset packages work with it, in which case the call to `connectAndUpdateWallet` wouldn't be made at all. Thoughts?

Reply:
Can you expand on that?  You mean make asset packages have working Connect after a Disconnect so that nothing needs to be reconstructed at this level or even inside of xcWallet?
Could happen, will require tons of mutex action in the asset.Wallet impls most likely.  Even trying this inside the xcWallet got nasty and it will be tougher inside each Wallet, but I'm open to that.
Will happily close this PR if there's a better solution.  I didn't find it.

Reply:
The opposite. Wallets packages should expect to be discarded after shutdown.

Reply:
Ahhh.  Lemme see about that.  I think the connect logic comes into play if you start core when the external wallet is down...

Reply:
There appears to be an assumption here that the wallet is disconnected after a proper shutdown, and therefore without any active state that might need to be rebuilt such as funding coins and queued redemption searches. We're also not replacing the `walletSet` of any existing orders that uses this wallet. It looks like the old way, in the case of an unintended disconnect, could potentially reconnect and resume trades. That wouldn't be the case anymore.

Reply:
Oh damn you're right the wallets are held in fields all over the place.